clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -761,9 +762,7 @@ private PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate> convertChangeEv
   }
 
   private V deserializeValueFromBytes(ByteBuffer byteBuffer, int valueSchemaId) {
-    Schema currentValueSchema = schemaReader.getValueSchema(valueSchemaId);
-    RecordDeserializer<V> valueDeserializer =
-        FastSerializerDeserializerFactory.getFastAvroGenericDeserializer(currentValueSchema, currentValueSchema);
+    RecordDeserializer<V> valueDeserializer = storeDeserializerCache.getDeserializer(valueSchemaId, valueSchemaId);
```

Comment:
Not related to this optimization, but...

Using the same schema for reader and writer is typically frowned upon... is this the right behavior? I am wondering if we should align it with the behavior of the GenericRecord thin client, which always upconverts to the latest schema (i.e. latest as the reader schema). Writer schema must be whatever the payload came with, of course.

Reply:
That makes sense, I can take that change in another PR (Ads team is a bit hot to try this out to determine if we need to do something more dramatic to get the performance better).  I'll file a BB

Reply:
LFG! 🚀